src/EventBus.js

```diff
@@ -1,100 +1,413 @@
-						listener.callback.apply(listener.scope, concatArgs);
-						numOfListeners += 1;
-					}
-				}
-			}
-		},
-		getEvents:function() {
-			var str = "";
-			for(var type in this.listeners) {
-				var numOfCallbacks = this.listeners[type].length;
-				for(var i=0; i<numOfCallbacks; i++) {
-					var listener = this.listeners[type][i];
-					str += listener.scope && listener.scope.className ? listener.scope.className : "anonymous";
-					str += " listen for '" + type + "'\n";
-				}
-			}
-			return str;
-		}
-	};
-	var EventBus = new EventBusClass();
-	return EventBus;
+    if (typeof exports === 'object' && typeof module === 'object')
+        module.exports = factory();
+    else if (typeof define === 'function' && define.amd)
+        define("EventBus", [], factory);
+    else if (typeof exports === 'object')
+        exports["EventBus"] = factory();
+    else
+        root["EventBus"] = factory();
+})(this, function () {
+
+    function iterator(array, callback) {
+        array = array || [];
+        var iterator = {
+            stop: function (result) {
+                this.isStopped = true;
+                this.result = result;
+            }
+        };
+        for (var i in array) {
+            callback(i, array[i], array, iterator);
+            if (iterator.isStopped)break;
+        }
+        if (iterator.isStopped)return iterator.result;
+    }
+
+    function slice(array, start, end) {
+        if (start > array.length)return [];
+        return [].slice.call(array, start == undefined ? 0 : start, end == undefined ? array.length : end) || [];
+    }
+
+    function processMultiTypes(type, callback, args) {
+        if (typeof type == "string" || type instanceof Array) {
+            var types = type;
+            if (typeof type == "string") types = type.trim().split(EventBusClass.DEFAULT.EVENT_TYPE_SPLIT_EXP);
+            if (types.length > 1) {
+                var that = this;
+                iterator(types, function (index, type) {
+                    args[0] = type;
+                    callback.apply(that, args);
+                });
+                return that;
+            } else {
+                type = types[0];
+            }
+        }
+        return type;
+    }
+    /**
+     * id generator
+     */
+    var nextId = (function () {
+        var _id = 1000;
+        return function () {
+            return ++_id;
+        }
+    })();
+
+    var EventBusClass = {};
+    EventBusClass = function () {
+        this.listeners = {};
+        this.regexListeners = [];
+    };
+    EventBusClass.prototype = {
+        /**
+         * EventBus.on("click",callback);
+         * EventBus.on("click",callback,buttonScope);
+         * @param type
+         * @param callback
+         * @param scope
+         * @returns {EventBusClass}
+         */
+        on: function (type, callback, scope) {
+            type = processMultiTypes.call(this, type, this.on, slice(arguments));
```

Comment:
Yes, perhaps this increases the readability of the code. ProcessMultiTypes method is to increase the unity of the EventBusClass.on and EventBusClass.off method to deal with type parameters, support for arrays or by comma and space separated string. If the return value equals EventBusClass object description has been processed, if no processing returns a single type processed by subsequent code.